cpp/kiss_icp/core/VoxelUtils.hpp

```diff
@@ -25,21 +25,25 @@
 
 #include <Eigen/Core>
 #include <cmath>
+#include <vector>
 
 namespace kiss_icp {
-
 using Voxel = Eigen::Vector3i;
-
 inline Voxel PointToVoxel(const Eigen::Vector3d &point, const double voxel_size) {
     return Voxel(static_cast<int>(std::floor(point.x() / voxel_size)),
                  static_cast<int>(std::floor(point.y() / voxel_size)),
                  static_cast<int>(std::floor(point.z() / voxel_size)));
 }
+/// Voxelize a point cloud keeping the original coordinates
+std::vector<Eigen::Vector3d> VoxelDownsample(const std::vector<Eigen::Vector3d> &point_cloud,
+                                             const double voxel_size);
 
-struct VoxelHash {
-    size_t operator()(const Voxel &voxel) const {
+}  // namespace kiss_icp
+
+template <>
```

Comment:
so we need to vote for partial template specialisation or custom type, it's 1-1 now, so it's @benemer choice now :)

Follow up of https://github.com/PRBonn/kiss-icp/pull/362#discussion_r1673016356

Reply:
I agree with @nachovizzo that using the `VoxelHash` struct makes reading a bit easier. However, I am not sure how "accessible" this hash function is in the first place. For example, it is not clear where it is coming from and what those numbers are. Given this, I think I am okay with "hiding" it and using the partial specialization.

Reply:
oki doki!